rest/route/agent.go

```diff
@@ -661,6 +661,29 @@ func (h *attachFilesHandler) Run(ctx context.Context) gimlet.Responder {
 		})
 	}
 
+	catcher := grip.NewBasicCatcher()
+	// If there are previous executions, update entries in case any content types have changed.
+	if t.Execution > 0 {
```

Comment:
I'm a little unclear on why this is necessary.  I don't think the task execution or id are relevant since the new task doesn't have any context into the original task. 
The way I understand the problem:
`task_a` runs for the first time and creates a placeholder called `file_a`.
`task_b` runs and then [uploads a new file in the same path of `file_a`](https://github.com/10gen/mongo/blob/7bf48bc837239b9b67526b4413e4d3ce7c186249/buildscripts/resmokelib/hang_analyzer/gen_hang_analyzer_tasks.py#L82-L92) effectively overwriting it on s3. 
`task_b` has no knowledge of what execution `task_a` is on nor does it necessarily know `task_a`s id. All it knows is its s3 file path.
`file_a` in `task_a` is now a different file but the `mime/content_type` type stored in the DB still refers to the old entry. 

This operation would need to match on the s3 file path as opposed to the task. Or allow the option to provide a different task id to update its attached files to associate the file with.

Reply:
I hesitate to allow a flow that allows tasks to modify content that has to do with other files (although I recognize that's what's happening in s3 anyway). I don't really expect multiple tasks to be touching the same files though, and writing a new query that looks at s3 path and bucket instead of task ID and execution is going to require a new query and a new index to keep it efficient, which I'm not sure is worth this given it's more of a nice-to-have.

I can add a comment to describe that this won't catch everything, but if you think this is insufficient then I think we should just close the ticket -- lmk what you think!

Reply:
The main goal was to provide an "official" way to update that original file from another task and ensure the `content_type` is kept in sync with the actual file in s3. But since we can't do that  I don't know just how beneficial it would be only to update the current task. Closing with a comment makes sense to me if there isn't another compelling reason to support the other user case you implemented.